### PR TITLE
Combined defs + tables + more

### PR DESCRIPTION
**Pull request by asutula**

### CODE REVIEW

File: packages/web/app/[team]/[project]/_components/sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                  <div className="ml-auto size-2 rounded-full bg-destructive" />
+                )}
+              </Button>
+            </Link>
+          );
+        })}
+        <div className="flex flex-1 items-center justify-center gap-x-2">
+          <TooltipProvider>
+            <Tooltip>
+              <TooltipTrigger asChild>
```
Comment (joewagner) [inline packages/web/app/[team]/[project]/_components/sidebar.tsx:177]: I don't see any errors, but this tooltip and the one below aren't working for me.

File: packages/web/app/[team]/[project]/[env]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+  const tables = deployments.map((d) => ({
+    chainId: d.deployment.chainId,
+    tableId: d.deployment.tableId,
+  }));
+
+  const chainTypes = new Set();
+  for (const table of tables) {
+    chainTypes.add(
+      chainsMap.get(table.chainId)?.testnet ? "testnet" : "mainnet",
```
Comment (joewagner) [inline packages/web/app/[team]/[project]/[env]/page.tsx:49]: It looks like this might result in hardhat being treated like a mainnet table.

File: packages/web/app/[team]/[project]/_components/sidebar.tsx (hunk 1 above)
Comment (asutula) [reply packages/web/app/[team]/[project]/_components/sidebar.tsx:177]: Fixing somehting related, let me push here in a minute and let's see what happens.

Comment (asutula) [reply packages/web/app/[team]/[project]/_components/sidebar.tsx]: Fixed.

File: packages/web/app/[team]/[project]/[env]/page.tsx (hunk 2 above)
Comment (asutula) [reply packages/web/app/[team]/[project]/[env]/page.tsx]: Fixed.

**Comment by dtbuchholz** [comment]
@asutula just added some comments on Vercel. i haven't used that feature before, it's pretty handy...but lmk if it didn't work.

generally, i love this design way more...sidebars make the nav much smoother! my main issue is re: real estate considerations wrt the "Tables" and listed tables. the Vercel comment describes it in more detail.

File: packages/web/app/[team]/(team)/layout.tsx [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
-    <div className="flex flex-1 flex-col">
-      <SubNavHeader>
-        <NavTeam />
-      </SubNavHeader>
-      {children}
+    <div className="flex flex-1 items-stretch">
+      <div className="sticky top-[3.55rem] h-[calc(100vh-3.55rem)] min-w-40 overflow-y-auto overflow-x-hidden border-r border-[#080A1E] bg-card">
+        <Sidebar />
+      </div>
+      <div className="w-full">{children}</div>
```
Comment (asutula) [inline packages/web/app/[team]/(team)/layout.tsx:13]: Added some css to make the overflow scrolling behavior of the sidebar vs the main content area work properly.

File: packages/web/app/[team]/[project]/[env]/[table]/page.tsx
Comment (asutula) [inline packages/web/app/[team]/[project]/[env]/[table]/page.tsx:1]: There was some refactoring to create and use shared components to display table information whether viewing a tableland table or a table that is part of a Studio project. This resulted in lots of files moved around and new components for shared code. The best thing is to just verify the app works as expected. It's just lots of react component code.

File: packages/web/app/[team]/[project]/[env]/page.tsx
Comment (asutula) [inline packages/web/app/[team]/[project]/[env]/page.tsx:1]: This is the project (environment) overview page.

File: packages/web/app/[team]/[project]/[env]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                </div>
+                <CardDescription>
+                  Your project includes both tables that have been deployed a
+                  mainnet and tables that have been deployed to a testnet. This
+                  is not recommended, as it can lead to unexpected behavior in
+                  Studio when querying information about your tables and data
+                  from your tables.
+                </CardDescription>
+              </CardHeader>
+            </Card>
```
Comment (asutula) [inline packages/web/app/[team]/[project]/[env]/page.tsx:72]: Added this alert if we detect an environment has tables deploed to a testnet and tables deployed to a mainnet.

File: packages/web/app/_components/primary-header-item.tsx
Comment (asutula) [inline packages/web/app/_components/primary-header-item.tsx:1]: I updated this component that displays the breadcrumb style team/project/(env coming soon) navigation in the upper right of Studio. The goal was for this component to also work if you're viewing teams/projects for teams that you don't belong to and for it to work if you are unauthenticated. 

The implementation is a little hard to follow because I tried to minimize the number of network requests required to gather all the needed data. The component is initialized with a list of user teams and the projects for each team. If we can find the team/project/projects for the page in that list, all good, no network requests needed. If not (because the user in unauthenticated or viewing a team/project they are not a member of) we make API requests to gather information about the team and project for the page.

File: packages/web/components/edit-def.tsx
Comment (asutula) [inline packages/web/components/edit-def.tsx:1]: This is the form for editing a table def. For now, it only supports editing the name and description. I ran into some difficulties when editing column definitions. It's solvable, but just complex and have to learn more about how to do it. Will address that in a future PR.

File: packages/web/components/profile.tsx
Comment (asutula) [inline packages/web/components/profile.tsx:1]: Noticed some not perfect layout in the main site header. Fixed that here in in some other related files.

File: packages/web/components/sidebar.tsx
Comment (asutula) [inline packages/web/components/sidebar.tsx:1]: Some reusable components that handle layout of any sidebar component that we create.

File: packages/web/components/sql-logs.tsx
Comment (asutula) [inline packages/web/components/sql-logs.tsx:1]: Updated our sql logs component to support displaying logs from one or many tables.

File: packages/web/components/table-menu.tsx
Comment (asutula) [inline packages/web/components/table-menu.tsx:1]: This table menu is displayed when viewing a tableland table or studio project table, and displays actions to take depending on the context (tableland table, studio table, deployed or not, user authed or not, etc).

File: packages/web/components/table-settings.tsx
Comment (asutula) [inline packages/web/components/table-settings.tsx:1]: A sheet to display table settings. This displays edit interface for the table def as well as deleting the table def and "undeploying" the table def.

File: packages/web/components/table-wrapper.tsx
Comment (asutula) [inline packages/web/components/table-wrapper.tsx:1]: A shared component that is used both when displaying a tableland table and a studio project table.

File: packages/web/components/team-switcher.tsx
Comment (asutula) [inline packages/web/components/team-switcher.tsx:1]: Updates to the team switcher to support new breadcrumb nav beahvior.

File: packages/web/components/undeploy-table-dialog.tsx
Comment (asutula) [inline packages/web/components/undeploy-table-dialog.tsx:1]: A dialog to confirm that the user wants to undeploy a table.

File: packages/web/lib/featured-projects.ts
Comment (asutula) [inline packages/web/lib/featured-projects.ts:1]: Improving the logic in here based on our new resolution of the base url.

File: packages/web/lib/validator-queries.ts [hunk 5]
```diff
@@ -77,15 +78,28 @@ export interface SqlLog {
   txHash: string;
 }
 
+// Tables should all be on mainnets or all be on testnets,
+// othwerwise the results will not be as expected.
```
Comment (asutula) [inline packages/web/lib/validator-queries.ts:82]: Important.

File: packages/web/lib/validator-queries.ts [hunk 6]
```diff
@@ -99,15 +113,16 @@ export async function getSqlLogs(
...
     (eventType = 'ContractCreateTable' OR eventType = 'ContractRunSQL')
   ORDER BY
-    blockNumber DESC, eventIndex DESC
+    timestamp DESC, e.chain_id ASC, blockNumber DESC, eventIndex DESC
   LIMIT ${limit}
   `;
 
-  const uri = encodeURI(`${baseUrlForChain(chain)}/query?statement=${query}`);
+  const uri = encodeURI(
+    `${baseUrlForChain(tables[0].chainId)}/query?statement=${query}`,
```
Comment (asutula) [inline packages/web/lib/validator-queries.ts:124]: This `tables[0]` to get the baseUrl is why that important comment above exists.

File: packages/web/lib/validator-queries.ts
Comment (asutula) [inline packages/web/lib/validator-queries.ts:1]: Updates to our validator query to support multiple tables when querying sql logs.

**Comment by asutula** [comment]
Haven't seen the tests pass in many pushes to github. Hmm.

**Comment by joewagner** [comment]
> Haven't seen the tests pass in many pushes to github. Hmm.

As per the comment in Discord, I can't say for sure what's wrong but the vercel kv store probably needs to be changed.  I don't have permission to do that here, but I can jump in voice and discuss if that's helpful.

File: packages/web/components/table-settings.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+                  Delete the {props.def.name} table from your project:
+                </p>
+                <Button variant="secondary" onClick={onDeleteTable}>
+                  Delete table
+                </Button>
+              </div>
+              {showUndeploy && (
+                <div className="space-y-2">
+                  <p className="text-sm font-medium">
+                    Un-deploy the {props.def.name} table:
```
Comment (joewagner) [inline packages/web/components/table-settings.tsx:115]: This lets me un-deploy a table that I imported.  I think allowing this might make sense, but when I go back to the project overview the table name in the sidebar shows it as being deployed.
I can click into the table and re-deploy it as normal, which is good imo.  This, interestingly, allows you to work around the bug that stops you from using a schema from another project.

**Review by joewagner — DISMISSED** [review]
@asutula I found a few things that don't quite work right. I put comments inline.

I'm approving this since the PR is pretty big and affects a lot of files. It might make sense to just approve, merge, and open tickets for the issues.
Doing this would allow us to rebase the other open PRs on top of this, which might help avoid conflicts.  

Alternatively, if you want to work through the issues I found before merging, I'm happy to help with that.

**Comment by asutula** [comment]
Great @joewagner, thank you! I'll read through those comments, fix everything that is easy/obvious, and open issues for anything else.

File: packages/web/components/sql-logs.tsx [hunk 8]
```diff
@@ -34,7 +32,7 @@ export default function SQLLogs({
         beforeTimestamp = undefined;
       }
       setLoading(true);
-      const res = await getSqlLogs(chain, tableId, pageSize, beforeTimestamp);
+      const res = await getSqlLogs(tables, pageSize, beforeTimestamp);
```
[the remark this answers is not in the document]
Comment (joewagner) [reply packages/web/components/sql-logs.tsx:35]: Ah, that makes sense.  I do see the alert, but I didn't realize it would affect the logs

[the remark this answers is not in the document]
Comment (asutula) [reply packages/web/components/sql-logs.tsx:35]: Yea, it's not an ideal solution, but easier/faster than making environments "mainnet" or  "testnet" and enforcing that. We can do that long term, though.

File: packages/web/components/table-menu.tsx [hunk 9]
```diff
@@ -1,62 +1,191 @@
...
+            </DropdownMenuItem>
+          )}
+          {displayImport && (
+            <DropdownMenuItem onSelect={() => setImportTableFormOpen(true)}>
+              Import table into Studio project
+            </DropdownMenuItem>
+          )}
           <DropdownMenuItem onSelect={() => setNewDefFormOpen(true)}>
-            Use table schema in Studio project
+            Use table definition in Studio project
```
[the remark this answers is not in the document]
Comment (asutula) [reply packages/web/components/table-menu.tsx:188]: Found the problem and fixed it.

File: packages/web/components/table-settings.tsx (hunk 7 above)
Comment (asutula) [reply packages/web/components/table-settings.tsx:115]: Fixed.

**Comment by asutula** [comment]
Ok @joewagner I fixed all the problems you found. Give this a quick review when you have a sec. Thx!

**Review by joewagner — APPROVED** [review]
@asutula Awesome, let's deploy!
